public/app/plugins/datasource/alertmanager/ConfigEditor.tsx

```diff
@@ -33,6 +33,19 @@ const IMPL_OPTIONS: Array<SelectableValue<AlertManagerImplementation>> = [
 export const ConfigEditor = (props: Props) => {
   const { options, onOptionsChange } = props;
 
+  // As we default to Mimir, we need to make sure the implementation is set from the start
+  useEffect(() => {
+    if (!options.jsonData.implementation) {
+      onOptionsChange({
+        ...options,
+        jsonData: {
+          ...options.jsonData,
+          implementation: AlertManagerImplementation.mimir,
+        },
+      });
```

Comment:
Are we able to use `produce` to make the change to the object instead?

Reply:
sure!

Reply:
done!